### PR TITLE
Headers with associated cpp are self sufficient

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

File: 04.build-guide/08.coding-standards/docs.md [hunk 1]
```diff
@@ -363,7 +363,27 @@ Include guards should be in the following format: hifi_$BASENAME_h.
 
 ```
 
-#### 2.2.2 Include statements should be sorted and grouped
+#### 2.2.2 Source files must include their associated header first
+
+The associated header is always included first to guarantee is it self-sufficient.
```
Comment (birarda) [inline 04.build-guide/08.coding-standards/docs.md]: it is

Comment (birarda) [reply 04.build-guide/08.coding-standards/docs.md]: Also we should still probably have them grouped by external library and our libraries

Comment (Atlante45) [reply 04.build-guide/08.coding-standards/docs.md]: > Also we should still probably have them grouped by external library and our libraries
What do you mean by that? I kept the groupings in my example.

Comment (birarda) [reply 04.build-guide/08.coding-standards/docs.md]: Right, but you changed the title of the section to be specific to what you wanted to add

**Pull request merged**